EventKit/eventkit-cloud
# Add loading spinner for providers

~NOTE: this PR is on top of @Coop56's geospatialFiltering branch. Many of the changes are from that branch.~

Display a loading spinner on create step 1 and 2
Refactor getProviders action to add context required for loading spinner  - it now emulates the getProviderTasks action
Fix tests + refactor to get rid of regressions. 

To test: 
1. open the coverages/basemaps tab
draw a small box on the map with only a few providers' data in it. Draw a big box on the map with lots of providers data in it, watch for a quick loading icon in the coverages + basemaps tabs when you finish drawing the big box

2. draw a small box on the map. Go to create step 2. Click on Sort and Filter. Toggle 'Selected Area' on and off and observe the loading icon in the providers list.  

3. Check for regressions.
